Lib/shelve.py

```diff
@@ -106,6 +106,19 @@ def __init__(self, dict, protocol=None, writeback=False,
             self.serializer = serializer
             self.deserializer = deserializer
 
+    @staticmethod
+    def _validate_serialized_value(serialized_value, original_value):
+        if (serialized_value is None or
+            not isinstance(serialized_value, (bytes, str))):
+            if serialized_value is None:
+                invalid_type = "None"
+            else:
+                invalid_type = type(serialized_value).__name__
+            msg = (f"Serializer returned {invalid_type} for value "
+                   f"{original_value!r} But database values must be "
```

Comment:
Nitpick, 'But' should be lowercase unless it follows a period:

```suggestion
                   f"{original_value!r}, but database values must be "
```

Adding a period before it is another option.